flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterDescriptor.java

```diff
@@ -867,27 +870,28 @@ private ApplicationReport startAppMaster(
 				TaskManagerOptions.TOTAL_PROCESS_MEMORY,
 				clusterSpecification.getTaskManagerMemoryMB() + "m");
 
-		// Upload the flink configuration
-		// write out configuration file
-		File tmpConfigurationFile = File.createTempFile(appId + "-flink-conf.yaml", null);
-		tmpConfigurationFile.deleteOnExit();
-		BootstrapTools.writeConfiguration(configuration, tmpConfigurationFile);
+		{
+			// Upload the flink configuration
+			// write out configuration file
+			File tmpConfigurationFile = File.createTempFile(appId + "-flink-conf.yaml", null);
+			tmpConfigurationFile.deleteOnExit();
```

Comment:
Since it has been manually deleted，i think "tmpConfigurationFile.deleteOnExit();" is not necessary，
and it's better to contain in finally{}

Reply:
Make sense. I force push a new commit.